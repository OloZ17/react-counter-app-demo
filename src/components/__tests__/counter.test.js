import React from 'react';
import { render, screen } from '@testing-library/react';
import Counter from '../counter';

describe('Testing Counter() Component', () => {
  const counter = { id: 1, value: 0 };

  const onIncrement = jest.fn();
  const onDecrement = jest.fn();
  const onDelete = jest.fn();

  it('should take a snapshot', () => {
    const { asFragment } = render(
      <Counter
        counter={counter}
        onIncrement={onIncrement}
        onDecrement={onDecrement}
        onDelete={onDelete}
      />
    );
    expect(asFragment(<Counter />)).toMatchSnapshot();
  });

  it('should render formatCount Zero with right warning class', () => {
    render(
      <Counter
        counter={counter}
        onIncrement={onIncrement}
        onDecrement={onDecrement}
        onDelete={onDelete}
      />
    );
    expect(screen.getByText('Zero')).toBeInTheDocument();
    expect(screen.getByText('Zero')).toHaveClass('badge m-2 bg-warning');
    expect(screen.getByTestId('decrement-button')).toHaveAttribute('disabled');
  });
  it('should render formatCount 1 with right primary class', () => {
    counter.value = 1;
    render(
      <Counter
        counter={counter}
        onIncrement={onIncrement}
        onDecrement={onDecrement}
        onDelete={onDelete}
      />
    );
    expect(screen.getByText('1')).toBeInTheDocument();
    expect(screen.getByText('1')).toHaveClass('badge m-2 bg-primary');
    expect(screen.getByTestId('decrement-button')).not.toHaveAttribute(
      'disabled'
    );
  });
});
