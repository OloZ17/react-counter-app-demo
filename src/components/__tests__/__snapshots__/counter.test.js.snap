// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Counter() Component should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="row justify-content-center"
    >
      <div
        class="col-4 text-center"
      >
        <span
          class="badge m-2 bg-warning"
          data-testid="total-counter"
          style="font-size: 21px; padding: 9px; vertical-align: middle;"
        >
          Zero
        </span>
        <button
          class="btn btn-secondary"
          data-testid="increment-button"
        >
          <i
            aria-hidden="true"
            class="fa fa-plus-circle"
          />
        </button>
        <button
          class="btn btn-info m-2"
          data-testid="decrement-button"
          disabled=""
        >
          <i
            aria-hidden="true"
            class="fa fa-minus-circle"
          />
        </button>
        <button
          class="btn btn-danger"
          data-testid="delete-button"
        >
          <i
            aria-hidden="true"
            class="fa fa-trash-o"
          />
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
